src-cpp/app/modules/shared/section_model.cpp

```diff
@@ -0,0 +1,124 @@
+#include "section_model.h"
+
+namespace Shared::Models
+{
+SectionModel::SectionModel(QObject* parent)
+    : QAbstractListModel(parent)
+{ }
+
+QHash<int, QByteArray> SectionModel::roleNames() const
+{
+    QHash<int, QByteArray> roles;
+
+    roles[Id] = "id";
+    roles[SectionType] = "sectionType";
+    roles[Name] = "name";
+    roles[AmISectionAdmin] = "amISectionAdmin";
+    roles[Description] = "description";
+    roles[Image] = "image";
+    roles[Icon] = "icon";
+    roles[Color] = "color";
+    roles[HasNotification] = "hasNotification";
+    roles[NotificationsCount] = "notificationsCount";
+    roles[Active] = "active";
+    roles[Enabled] = "enabled";
+    roles[Joined] = "joined";
+    roles[IsMember] = "isMember";
+    roles[CanJoin] = "canJoin";
+    roles[CanManageUsers] = "canManageUsers";
+    roles[CanRequestAccess] = "canRequestAccess";
+    roles[Access] = "access";
+    roles[EnsOnly] = "ensOnly";
+    roles[MembersModel] = "members";
+    roles[PendingRequestsToJoinModel] = "pendingRequestsToJoin";
+    return roles;
+}
+
+int SectionModel::rowCount(const QModelIndex& parent = QModelIndex()) const
+{
+    return m_items.size();
+}
+
+QVariant SectionModel::data(const QModelIndex& index, int role) const
+{
+    if(!index.isValid())
+    {
+        return QVariant();
+    }
+
+    if(index.row() < 0 || index.row() > m_items.size())
```

Comment:
`index.row() >= m_items.size()` cos `m_items.at(m_items.size())` is out of range